File: src/core/PipelineOutput.ts
import IOTypes from './IOTypes.js'
import Image from './Image.js'
import Mesh from './Mesh.js'
import PolyData from './vtkPolyData.js'

interface PipelineOutput {
  path: string;
  type: typeof IOTypes[keyof typeof IOTypes];
  data?: string | Uint8Array | Image | Mesh | PolyData;
}

export default PipelineOutput

File: packages/core/python/itkwasm/itkwasm/pipeline_output.py
from dataclasses import dataclass
from typing import Optional, Union

from .interface_types import InterfaceTypes
from .text_file import TextFile
from .text_stream import TextStream
from .binary_file import BinaryFile
from .binary_stream import BinaryStream
from .image import Image
from .mesh import Mesh
from .polydata import PolyData

@dataclass
class PipelineOutput:
    type: InterfaceTypes
    data: Optional[Union[str, bytes, TextStream, BinaryStream, TextFile, BinaryFile, Image, Mesh, PolyData]] = None
    path: Optional[str] = None
